x11-misc/polybar/polybar-3.5.0.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8,9} )
+XPP_COMMIT="044e69d05db7f89339bda1ccd1efe0263b01c8f6"
+I3IPCPP_COMMIT="cb008b30fc5f3febfe467884cb0211ee3c16386b"
+
+inherit cmake python-single-r1
+
+DESCRIPTION="A fast and easy-to-use tool for creating status bars"
+HOMEPAGE="https://github.com/polybar/polybar"
+SRC_URI="https://github.com/polybar/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/polybar/xpp/archive/${XPP_COMMIT}.tar.gz -> xpp-${XPP_COMMIT}.tar.gz
+	https://github.com/polybar/i3ipcpp/archive/${I3IPCPP_COMMIT}.tar.gz -> i3ipcpp-${I3IPCPP_COMMIT}.tar.gz"
+
+KEYWORDS="~amd64 ~x86"
+LICENSE="MIT"
+SLOT="0"
+IUSE="alsa curl i3wm ipc mpd network pulseaudio"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="
+	${PYTHON_DEPS}
+	$(python_gen_cond_dep 'x11-base/xcb-proto[${PYTHON_MULTI_USEDEP}]')
+	x11-libs/cairo[X,xcb(+)]
+	x11-libs/libxcb[xkb]
+	x11-libs/xcb-util-image
+	x11-libs/xcb-util-wm
+	x11-libs/xcb-util-xrm
+	alsa? ( media-libs/alsa-lib )
+	curl? ( net-misc/curl )
+	i3wm? (
+		dev-libs/jsoncpp:=
+		|| ( x11-wm/i3 x11-wm/i3-gaps )
+	)
+	mpd? ( media-libs/libmpdclient )
+	network? ( net-wireless/wireless-tools )
+	pulseaudio? ( media-sound/pulseaudio )
+"
+
+RDEPEND="${DEPEND}"
+
+src_prepare() {
+	rmdir "${S}"/lib/xpp || die
+	mv "${WORKDIR}"/xpp-$XPP_COMMIT "${S}"/lib/xpp || die
+
+	rmdir "${S}"/lib/i3ipcpp || die
+	mv "${WORKDIR}"/i3ipcpp-$I3IPCPP_COMMIT "${S}"/lib/i3ipcpp || die
```

Comment:
perhaps this should be done in src_unpack?

Reply:
That doesn't unpack anything, so probably not.

Reply:
Could you unpack those sources right into `"${S}"/lib/{xpp,i3ipcpp}`? Doesn't look like `unpack` takes any options though. Still, I feel like src_unpack suits this better as it is moving items from SRC_URI into the required place rather than just editing a few lines or patching something

Reply:
Preparing things for compilation counts as preparation far more than unpacking. Putting this in src_unpack would seem to result in more lines being added to the ebuild than keeping it how it is.